pages/react-canvas-components.mdx

```diff
@@ -0,0 +1,256 @@
+  }}
+>
+  <RandomHexagons isAnimating={true} />
+</div>
+
+A single hexagon is defined with the folowing data:
+
+- A position on the screen, defined by two `x` and `y` number values.
+- A `radius` that defines its size.
+- A `rotation` so that all hexagons don't look aligned.
+- A `color`.
+
+We need a function that is able to generate some random hexagons to display. The randomisation code is not relevant here, let's assume we have a way to get an array of hexagons. As for how to draw the shape of an hexagon, we need to draw a line between all its corners, and then fill it with a color:
+
+```jsx
+// This article explains all the math behind hexagons
+// https://www.redblobgames.com/grids/hexagons/
+const corners = getHexagonCorners(x, y, radius, rotation);
+
+context.beginPath();
+corners.forEach((corner, index) => {
+  if (index === 0) {
+    context.moveTo(corner.x, corner.y);
+  } else {
+    context.lineTo(corner.x, corner.y);
+  }
+});
+context.fillStyle = color;
+context.fill();
+```
+
+Now, how could we extract this logic into its own `Hexagon` component ? We need the canvas' context to draw anything on it. We could pass it to all the child components via a prop, but this approach can be a bit tedious when children are nested. Another way of doing this is by using a [React context](https://reactjs.org/docs/context.html) to share "global" values between components.
+
+## 📦 A context in a context
+
+At this point the naming gets a bit tricky as we are trying to share a canvas' _context_ via a React _context_. Once we create a React context, we need to use the context's `Provider` to share a value. In the case of our `Canvas` component, which would look like this:
+
+```jsx hexagons-30dkf src/Canvas.js
+// we create a React context with a _null_ default value
+const SharingContext = React.createContext(null);
+
+const Canvas = props => {
+  const canvasRef = React.useRef(null);
+  const [renderingContext, setRenderingContext] = React.useState(null);
+
+  // the canvas rendering context is not immediately avalaible
+  // as the canvas node first needs to be added to the DOM by react
+  React.useEffect(() => {
+    const context2d = canvasRef.current.getContext('2d');
+    setRenderingContext(context2d);
+  }, []);
+
+  return (
+    <SharingContext.Provider value={renderingContext}>
+      <canvas ref={canvasRef} />
+      {/* the hexagons are passed through the `children` prop */}
+      {props.children}
+    </SharingContext.Provider>
+  );
+};
+```
+
+The `Hexagon` component needs to consume this React context to read its value – here with the `useContext` hook.
+
+```jsx hexagons-30dkf src/Hexagon.js
+const Hexagon = props => {
+  // we get the rendering context by comsuming the React context
+  const renderingContext = React.useContext(SharingContext);
+
+  if (renderingContext !== null) {
+    // hexagon drawing logic
+  }
+};
+```
+
+Now, both our `Canvas` and `Hexagon` components are ready. We can display several random hexagons:
```

Comment:
- Remove the comma
- "Now that both our components are ready we are able to display randomly-generated hexagons"